doc/csharp/csharp-null-ref-types-proposal.md

```diff
@@ -0,0 +1,170 @@
+comment in the code disables it. No nullable checks are made and no compiler 
+warnings are generated for this code, even if the code is compiled in a project 
+with nullable checks enabled.
+
+### 2. .NET Framework projects default to using C# 7.3
+
+*.NET Framework* projects default to using compiler version C# 7.3 which does
+not support nullable reference type features.
+
+*Visual Studio 2022* does not let you change to a later compiler version for
+*.NET Framework* projects:
+- In *Advanced Build Settings | Language version* the drop down to select the 
+language version is greyed out and says “*Automatically selected based on 
+framework version*”
+
+See https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version 
+
+Note: It is possible to edit the project file and explicitly set the language 
+version to a later version, but other language features that require runtime 
+changes in the CLR will not work. C# 7.3 is the latest language version that is 
+fully supported with *.NET Framework* projects.
+
+### 3. Building gRPC projects with nullable enabled works today
+
+There does not appear to be a problem building *.NET Core* projects that have
+nullable enabled and are using the generated code. All the `grpc/dotnet-grpc`
+examples already have nullable context enabled and they build without errors 
+or warnings.
+
+### 4. Breaking backwards compatibility with gRPC service classes
+
+The generated code for gRPC services could have attributes added to specify that 
+a non-null message or stream must be passed to a service, and a non-null value 
+is returned.
+
+However this can break existing user code. Adding these attributes to the base 
+class may cause errors (not warnings) in user code that has already been derived 
+from the base class:
+
+> *error CS8765: Nullability of type of parameter 'request' doesn't match 
+overridden member (possibly because of nullability attributes).*
+
+Note: the advantages of adding these attributes to the service methods is 
+questionable as the only caller of these methods is the server hosting the 
+services, not user code. Therefore this may not be an issue if it is not needed 
+to be done.
+
+## Disadvantages of this propsal
+
+There are a few places where adding annotations or attributes to the generated
+code to help with the static flow analysis could be useful. These are:
+- Checking non-null setters for certain fields in a message at compile time,
+ such as fields that are strings. Runtime checks are still done in the
+ generated code.
+- Checking a non-null message or stream is passed to a gRPC service call.
+- Specifying a non-null message or stream is returned from a gRPC service call.
+
+By not supporting nullable reference types in the generated code these compile
+time checks cannot be done.
+
+## Outside the scope of this proposal
+
+### Changes to related packages
+
+Changes to `Google.Protobuf` library and other libraries 
```

Comment:
Possibly explicitly say that there is a long-standing (slow-moving) effort to enable NRT for Google.Protobuf?

Reply:
I think it is worth mentioning that well known types are in Google.Protobuf. Annotating them is valuable to the user experience.

It would be odd if user gRPC types were annotated, but `Goole.Protobuf.Any` wasn't.